learning_observer/learning_observer/kvs.py

```
@@ -286,26 +287,37 @@ def kvs_startup_check():
 
     Checks like this one allow us to fail on startup, rather than later
     '''
-    global KVS
+    global KVS, KVS_DICT
     try:
         KVS_MAP = {
             'stub': InMemoryKVS,
             'redis_ephemeral': EphemeralRedisKVS,
             'redis': PersistentRedisKVS
         }
-        KVS = KVS_MAP[learning_observer.settings.settings['kvs']['type']]
+        # register KVSs
+        for kvs_item in learning_observer.settings.settings['kvs']:
+            kvs_type = kvs_item['type']
+            kvs_class = KVS_MAP[kvs_type]
+            # set timeout for ephemeral redis connections
+            if kvs_type == 'redis_ephemeral':
+                kvs_class = functools.partial(kvs_class, kvs_item['expiry'])
+            KVS_DICT[kvs_item['name']] = kvs_class
+
+        # Set default KVS to not break legacy code
+        KVS = KVS_DICT.get('main', InMemoryKVS)
```

This should very much *not* have a default like this. 

If a deployed system falls into a default like this -- Collin's deploy at NCSU by upgrading, or a new setup, it *will* lose data on each restart. 

On code like this, you want to fail early.